### PR TITLE
chore: add basic mergify config

### PR DESCRIPTION
**Pull request by DanielMSchmidt**

### CODE REVIEW

**Comment by team-tf-cdk** [comment]
#### [`repos`] Terraform Plan 📖`success`

<a href="https://app.terraform.io/app/cdktf-team/prebuilt-providers/runs/run-9MJCNaGHAJ5n7meY" target="_blank">Remote Plan Link ↗️</a>

<details><summary>Show Plan</summary>

```tf
[command]/home/runner/work/_temp/93434310-707a-408f-af55-5773e4eb39c2/terraform-bin plan -no-color
Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/cdktf-team/prebuilt-providers/runs/run-9MJCNaGHAJ5n7meY

Waiting for the plan to start...

Terraform v1.0.7
on linux_amd64
Initializing plugins and modules...

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # github_team_repository.repos_cdktfproviderproject_managingteam_86F83671 will be created
  + resource "github_team_repository" "repos_cdktfproviderproject_managingteam_86F83671" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "admin"
      + repository = "cdktf-provider-project"
      + team_id    = "3949435"
    }

  # github_team_repository.repos_cdktfrepositorymanager_managingteam_31E92624 will be created
  + resource "github_team_repository" "repos_cdktfrepositorymanager_managingteam_31E92624" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "admin"
      + repository = "cdktf-repository-manager"
      + team_id    = "3949435"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

::debug::Terraform exited with code 0.
::debug::stderr: 
::debug::exitcode: 0


::set-output name=stderr::

::set-output name=exitcode::0

```

</details>

*Pusher: @DanielMSchmidt, Action: `pull_request`, Working Directory: ``, Workflow: `Plan`*

**Comment by team-tf-cdk** [comment]
#### [`repos-official-new`] Terraform Plan 📖`success`

<a href="https://app.terraform.io/app/cdktf-team/prebuilt-providers-official-new/runs/run-q9p5mAMngZsMe54i" target="_blank">Remote Plan Link ↗️</a>

<details><summary>Show Plan</summary>

```tf
[command]/home/runner/work/_temp/1b28057b-6265-476b-89db-0c0537423698/terraform-bin plan -no-color
Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/cdktf-team/prebuilt-providers-official-new/runs/run-q9p5mAMngZsMe54i

Waiting for the plan to start...

Terraform v1.3.1
on linux_amd64
Initializing plugins and modules...
data.github_team.reposofficialnew_cdktfteamcdktf_4A2C9F8E: Reading...
data.github_team.reposofficialnew_cdktfteamhashicorp_2FBDEDDB: Reading...
data.github_team.reposofficialnew_cdktfteamhashicorp_2FBDEDDB: Read complete after 1s [id=3949435]
data.github_team.reposofficialnew_cdktfteamcdktf_4A2C9F8E: Read complete after 10s [id=6602018]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.
::debug::Terraform exited with code 0.
::debug::stderr: 
::debug::exitcode: 0


::set-output name=stderr::

::set-output name=exitcode::0

```

</details>

*Pusher: @DanielMSchmidt, Action: `pull_request`, Working Directory: ``, Workflow: `Plan`*

**Comment by team-tf-cdk** [comment]
#### [`repos`] Terraform Plan 📖`success`

<a href="https://app.terraform.io/app/cdktf-team/prebuilt-providers/runs/run-aN2x7Cqj2hUS3mJp" target="_blank">Remote Plan Link ↗️</a>

<details><summary>Show Plan</summary>

```tf
[command]/home/runner/work/_temp/5b4baa2b-13dc-4e1c-9d26-cc3e221f016b/terraform-bin plan -no-color
Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/cdktf-team/prebuilt-providers/runs/run-aN2x7Cqj2hUS3mJp

Waiting for the plan to start...

Terraform v1.0.7
on linux_amd64
Initializing plugins and modules...

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # github_team_repository.repos_cdktfproviderproject_managingteam_86F83671 will be created
  + resource "github_team_repository" "repos_cdktfproviderproject_managingteam_86F83671" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "admin"
      + repository = "cdktf-provider-project"
      + team_id    = "3949435"
    }

  # github_team_repository.repos_cdktfrepositorymanager_managingteam_31E92624 will be created
  + resource "github_team_repository" "repos_cdktfrepositorymanager_managingteam_31E92624" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "admin"
      + repository = "cdktf-repository-manager"
      + team_id    = "3949435"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

::debug::Terraform exited with code 0.
::debug::stderr: 
::debug::exitcode: 0


::set-output name=stderr::

::set-output name=exitcode::0

```

</details>

*Pusher: @DanielMSchmidt, Action: `pull_request`, Working Directory: ``, Workflow: `Plan`*

**Comment by team-tf-cdk** [comment]
#### [`repos-official-new`] Terraform Plan 📖`success`

<a href="https://app.terraform.io/app/cdktf-team/prebuilt-providers-official-new/runs/run-7TjVM715nQ8pJbtJ" target="_blank">Remote Plan Link ↗️</a>

<details><summary>Show Plan</summary>

```tf
[command]/home/runner/work/_temp/9493f139-b842-41ad-91d1-eac128c7c602/terraform-bin plan -no-color
Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/cdktf-team/prebuilt-providers-official-new/runs/run-7TjVM715nQ8pJbtJ

Waiting for the plan to start...

Terraform v1.3.1
on linux_amd64
Initializing plugins and modules...
data.github_team.reposofficialnew_cdktfteamcdktf_4A2C9F8E: Reading...
data.github_team.reposofficialnew_cdktfteamhashicorp_2FBDEDDB: Reading...
data.github_team.reposofficialnew_cdktfteamhashicorp_2FBDEDDB: Read complete after 2s [id=3949435]
data.github_team.reposofficialnew_cdktfteamcdktf_4A2C9F8E: Read complete after 8s [id=6602018]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.
::debug::Terraform exited with code 0.
::debug::stderr: 
::debug::exitcode: 0


::set-output name=stderr::

::set-output name=exitcode::0

```

</details>

*Pusher: @DanielMSchmidt, Action: `pull_request`, Working Directory: ``, Workflow: `Plan`*

**Comment by team-tf-cdk** [comment]
#### [`repos`] Terraform Plan 📖`success`

<a href="https://app.terraform.io/app/cdktf-team/prebuilt-providers/runs/run-qthb35Z7aa3NUfGU" target="_blank">Remote Plan Link ↗️</a>

<details><summary>Show Plan</summary>

```tf
[command]/home/runner/work/_temp/d54caebb-5376-4100-94d2-e8a42a0112b7/terraform-bin plan -no-color
Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/cdktf-team/prebuilt-providers/runs/run-qthb35Z7aa3NUfGU

Waiting for the plan to start...

Terraform v1.0.7
on linux_amd64
Initializing plugins and modules...

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # github_team_repository.repos_cdktfproviderproject_managingteam_86F83671 will be created
  + resource "github_team_repository" "repos_cdktfproviderproject_managingteam_86F83671" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "admin"
      + repository = "cdktf-provider-project"
      + team_id    = "3949435"
    }

  # github_team_repository.repos_cdktfrepositorymanager_managingteam_31E92624 will be created
  + resource "github_team_repository" "repos_cdktfrepositorymanager_managingteam_31E92624" {
      + etag       = (known after apply)
      + id         = (known after apply)
      + permission = "admin"
      + repository = "cdktf-repository-manager"
      + team_id    = "3949435"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

::debug::Terraform exited with code 0.
::debug::stderr: 
::debug::exitcode: 0


::set-output name=stderr::

::set-output name=exitcode::0

```

</details>

*Pusher: @DanielMSchmidt, Action: `pull_request`, Working Directory: ``, Workflow: `Plan`*

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        method: squash
+        name: default
+        commit_message_template: |-
+          {{ title }} (#{{ number }})
+
+          {{ body }}
+    conditions:
+      - "#approved-reviews-by>=1"
+      - -label~=(do-not-merge)
+      - status-success=terraform
```
Comment (ansgarm) [inline .mergify.yml:21]: Do we need to include `license/cla` or is that somehow included by default?
Else we'd need to always remember to not approve PRs that have no cla approval.

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+pull_request_rules:
+  - name: Automatic merge on approval and successful build
+    actions:
+      delete_head_branch: {}
+      queue:
+        method: squash
```
Comment (ansgarm) [inline .mergify.yml:12]: nit: I'm a bit torn here, as I can see individual commits as helpful sometimes – but as this repo gets quite small changes usually, I'm fine with squashing I think.

**Comment by team-tf-cdk** [comment]
#### [`repos-official-new`] Terraform Plan 📖`success`

<a href="https://app.terraform.io/app/cdktf-team/prebuilt-providers-official-new/runs/run-KXdUv23bhMiZTHKt" target="_blank">Remote Plan Link ↗️</a>

<details><summary>Show Plan</summary>

```tf
[command]/home/runner/work/_temp/cf9f3cc1-d415-4ee0-b042-4b486b6caf9c/terraform-bin plan -no-color
Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/cdktf-team/prebuilt-providers-official-new/runs/run-KXdUv23bhMiZTHKt

Waiting for the plan to start...

Terraform v1.3.1
on linux_amd64
Initializing plugins and modules...
data.github_team.reposofficialnew_cdktfteamcdktf_4A2C9F8E: Reading...
data.github_team.reposofficialnew_cdktfteamhashicorp_2FBDEDDB: Reading...
data.github_team.reposofficialnew_cdktfteamhashicorp_2FBDEDDB: Read complete after 0s [id=3949435]
data.github_team.reposofficialnew_cdktfteamcdktf_4A2C9F8E: Read complete after 7s [id=6602018]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.
::debug::Terraform exited with code 0.
::debug::stderr: 
::debug::exitcode: 0


::set-output name=stderr::

::set-output name=exitcode::0

```

</details>

*Pusher: @DanielMSchmidt, Action: `pull_request`, Working Directory: ``, Workflow: `Plan`*

**Pull request merged**